lightning/src/ln/channel.rs

```diff
@@ -11060,6 +11101,15 @@ impl<'a, 'b, 'c, ES: Deref, SP: Deref> ReadableArgs<(&'a ES, &'b SP, &'c Channel
 			&channel_parameters.counterparty_parameters.as_ref()
 				.expect("Persisted channel must have counterparty parameters").pubkeys.revocation_basepoint);
 
+		let interactive_tx_signing_session = if channel_state.is_interactive_signing() {
+			interactive_tx_signing_session.zip(funding_transaction.as_ref()).map(|(mut session, funding_tx)| {
+				session.update_from_funding_tx_after_read(funding_tx);
+				session
+			})
+		} else {
+			None
+		};
```

Comment:
Instead of needing `update_from_funding_tx_after_read`, let's make `InteractiveTxSigningSession` implement `ReadableArgs` taking `Option<&Transaction>`. Then, in the macro, it would be something like:

```
(55, interactive_tx_signing_session, (option: ReadableArgs, funding_transaction.as_ref()))
```